### DISCUSSION TITLE
Adding Drawing part image prevents the cell update

### DISCUSSION THREAD

**Discussion by ZubairImran** [OfficeDev/Open-XML-SDK#1823] [General]
Hi Team, 
I am adding a image using drawing layer(drawing part) in a Excel file. I could able to add the image but the image prevents the cells available in the  background. Is there anyway to send the image to background using openxml sdk. Any pointer would be great help.

**Comment by mikeebowen** [comment]
Hi @ZubairImran,

This will add an image to the background of a sheet:

```csharp
using DocumentFormat.OpenXml.Packaging;

string xlsxPath = @"path/to/your.xlsx";
string imagePath = @"path/to/your/image.jpg";

using (var doc = SpreadsheetDocument.Open(xlsxPath, true))
{
    if (doc?.WorkbookPart is not { } workbookPart ||
        workbookPart.WorksheetParts.FirstOrDefault() is not { } worksheetPart)
    {
        throw new ArgumentException("Can't find worksheet part");
    }

    ImagePart imagePart = worksheetPart.AddImagePart("image/jpeg");
    var id = workbookPart.GetIdOfPart(imagePart);

    using (var stream = File.OpenRead(imagePath))
    {
        imagePart.FeedData(stream);
    }

    worksheetPart.Worksheet.AddChild(new DocumentFormat.OpenXml.Spreadsheet.Picture() { Id = id});
}
```

**Reply by ZubairImran** [reply]
@mikeebowen , Thanks for the above code and time. But when i tried it is not working and corrupting the excel. Am i missing anything.

```csharp
using (SpreadsheetDocument spreadsheet = SpreadsheetDocument.Open(filePath, true))
{
    WorkbookPart workbookPart = spreadsheet.WorkbookPart;
    WorksheetPart worksheetPart = GetWorksheetPartByName(workbookPart, sheetName);

    if (worksheetPart == null)
    {
        throw new Exception($"Worksheet '{sheetName}' not found.");
    }

    ImagePart imagePart = worksheetPart.AddImagePart("image/png");
    var id = worksheetPart.GetIdOfPart(imagePart);
    string imagePath = "c:\\users\\public\\t.png";
    using (var stream = File.OpenRead(imagePath))
    {
        imagePart.FeedData(stream);
    }

    worksheetPart.Worksheet.AddChild(new DocumentFormat.OpenXml.Spreadsheet.Picture() { Id = id });
    worksheetPart.Worksheet.Save();

}
 private static WorksheetPart GetWorksheetPartByName(WorkbookPart workbookPart, string sheetName)
{
    foreach (Sheet sheet in workbookPart.Workbook.Sheets)
    {
        if (sheet.Name == sheetName)
        {
            return (WorksheetPart)workbookPart.GetPartById(sheet.Id);
        }
    }
    return null;
}
```

**Reply by ZubairImran** [reply]
@mikeebowen , i have tried again with simple xlsx with single sheet, the above code is working fine. But its adding the image multiple time in same sheet. Is it the actual behaviour?

**Reply by mikeebowen** [reply]
Hi @ZubairImran, repeating the image in the background is the expected behavior, because a sheet has an unlimited number of rows, i.e. infinite height, there is no way for Excel to know how to scale the image.

**Reply by ZubairImran** [reply]
Thanks @mikeebowen for the prompt reply.

**Comment by ZubairImran** [comment]
Hi Bown,
Do Open-XML-SDK supports to keep only one background image meaning can we
delete the above inserted images(by iterating and keep only one)?

**Reply by mikeebowen** [reply]
Hi @ZubairImran, Do you mean you only want to show one background image and then have no background image below that first background image?

**Comment by ZubairImran** [comment]
Hi Bowen,
Thanks for your reply. I want to keep.only one background image in a sheet.

Thanks
Hyder

**Comment by mikeebowen** [comment]
Unfortunately, you can't stop background image tiling with Excel. This [Stack Exchange post](https://superuser.com/questions/1729559/how-to-add-a-background-image-to-only-specific-range-of-cells-in-excel) gives a possible work around with shapes, but I have not tried it.

**Comment by ZubairImran** [comment]
Hi Bowen ,
  Thanks for the support.  Any hint on the below would be a great help

**Reply by mikeebowen** [reply]
Hi @ZubairImran, as I explained in my earlier post, Excel does not provide a way to prevent background image tiling. I did find this [Stack Overflow post](https://superuser.com/questions/1729559/how-to-add-a-background-image-to-only-specific-range-of-cells-in-excel) that contains a possible workaround, but I have not tried to implement it, so I can't guarantee that it works.

**Comment by ZubairImran** [comment]
Hi  Bowen,
Thanks a lot. I will go through this. Thanks again
